packages/aws-cdk-lib/aws-secretsmanager/lib/rotation-schedule.ts

```diff
@@ -325,14 +325,21 @@ export class HostedRotation implements ec2.IConnectable {
       this.masterSecret.denyAccountRootDelete();
     }
 
+    let masterSecretArn: string | undefined;
+    if (this.masterSecret?.secretFullArn) {
+      masterSecretArn = this.masterSecret.secretArn;
+    } else if (this.masterSecret) { // ISecret as an imported secret with partial ARN
+      masterSecretArn = this.masterSecret.secretArn + '-??????';
+    }
```

Comment:
- `if (this.masterSecret?.secretFullArn)`
  - If we generate a secret construct for `this.masterSecret` in the stack, here is true.
  - Otherwise `this.masterSecret` is generated from `fromSecretArn()`, `fromSecretCompleteArn()` methods, etc. They also have a full ARN.
    - This `if` statement will be true even if the arg (so it is the arn) for the methods is a token or not.
- `else if (this.masterSecret)`
  - The secret is `ISecret` as an imported secret with partial ARN. It is generated from `fromSecretNameV2()`.
  - The `secretArn` in this block will be a token. But processing this string is not a problem because the CloudFormation template can use `Fn::Join` to combine them properly. (see the unit tests.)